contracts/TopShot.cdc

```diff
@@ -1191,7 +1224,7 @@ pub contract TopShot: NonFungibleToken {
 
         // lock takes a token id and a duration in seconds and locks
         // the moment for that duration
-        pub fun lock(id: UInt64, duration: UFix64) {
+        access(NonFungibleToken.Update | NonFungibleToken.Owner) fun lock(id: UInt64, duration: UFix64) {
```

Comment:
if locking/unlocking is considered an update of the NFT, I would consider using [the `NonFungibleToken.emitNFTUpdated()` function](https://github.com/onflow/flow-nft/blob/standard-v2/contracts/NonFungibleToken.cdc#L66) to emit the standard event to signify when a token is updated.

You can just borrow an entitled reference to the locked NFT here and use it to call that function.

Reply:
Can you include this event emission here?